### ISSUE TITLE
Constraints in methods requires `@template` in `jsdoc/require-template` rule

### ISSUE THREAD

**Issue by alexander-akait** [gajus/eslint-plugin-jsdoc#1648]
## Expected behavior

Do not ask about adding `@template K`

## Actual behavior

Asking about it

## ESLint Config

```js
{
  "jsdoc/require-template": ["error"]
}
```

## ESLint sample

```js
/**
 * @typedef {object} CodeGenMapOverloads
 * @property {<K extends string>(key: K) => CodeGenValue<K> | undefined} get
 * @property {<K extends string>(key: K, value: CodeGenValue<K>) => CodeGenerationResultData} set
 * @property {<K extends string>(key: K) => boolean} has
 * @property {<K extends string>(key: K) => boolean} delete
 */

/**
 * @typedef {Omit<Map<string, EXPECTED_ANY>, "get" | "set" | "has" | "delete"> & CodeGenMapOverloads} CodeGenerationResultData
 */
```

the same code using typescript:
```ts
export interface CodeGenerationResultData extends Omit<Map<string, any>, "get" | "set" | "has" | "delete"> {
    get<K extends string>(key: K): CodeGenValue<K> | undefined;
    
    set<K extends string>(key: K, value: CodeGenValue<K>): this;
    
    has<K extends string>(key: K): boolean;
    
    delete<K extends string>(key: K): boolean;
}
```

as you can see no `K` generic in types for `CodeGenerationResultData`

## Environment

- Node version: v24.11.1
- ESLint version v9.39.2
- `eslint-plugin-jsdoc` version: latest
